clientdb/sidecar.go

```diff
@@ -0,0 +1,173 @@
+package clientdb
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightninglabs/pool/sidecar"
+	"go.etcd.io/bbolt"
+)
+
+var (
+	// ErrNoSidecar is the error returned if no sidecar with the given
+	// multisig pubkey exists in the store.
+	ErrNoSidecar = errors.New("no sidecar found")
+
+	// sidecarsBucketKey is a bucket that contains all sidecars that are
+	// currently pending or completed. This bucket is keyed by the ticket ID
+	// and offer signing pubkey of a sidecar.
+	sidecarsBucketKey = []byte("sidecars")
+)
+
+const (
+	// sidecarKeyLen is the length of a sidecar ticket's key. It is the
+	// length of the sidecar ID (8 bytes) plus the length of a compressed
+	// public key (33 bytes).
+	sidecarKeyLen = 8 + 33
```

Comment:
👍 

Makes things a bit longer in practice, but nice to not have to worry about collisions. 

As an aside, I wonder if we should also add an _expiry_ to the tickets as well. Can be done easily later since the entire thing is a TLV record.